### ISSUE TITLE
Headings inside legends not recognised by JAWS

### ISSUE THREAD

**Issue by andymantell** [nhsuk/nhsuk-frontend#778]
# Bug Report

## What is the issue?

When headings (e.g. &lt;h1&gt;) are nested inside a &lt;legend&gt; JAWS does not announce them when pressing the `H` key.

I note that a _very_ similar issue has been closed in the past at https://github.com/nhsuk/nhsuk-frontend/issues/534. However this issue is current and is unrelated to any `overflow: hidden` css. It occurs in a plain html page outside of the nhsuk-frontend styles.

We have observed this in user research as well as my own testing. In user research it resulted in a fair amount of confusion for one particular user who regularly used the `H` key - often as the first thing they pressed after the page loaded.

The issue is currently being tracked in the following issues, but I wanted to open this one here so that it doesn't fly under the radar for us.

https://github.com/FreedomScientific/VFO-standards-support/issues/549
https://github.com/FreedomScientific/VFO-standards-support/issues/100
https://github.com/alphagov/govuk-frontend/issues/1605
https://github.com/w3c/html-aria/issues/350
https://github.com/hmrc/accessibility/issues/34

## What steps are required to reproduce the issue?

Create a page with the following markup, and listen to it with JAWS. Press the `H` key - observe that it says "No headings available"

```
<form>
  <fieldset>
    <legend><h1>This is a heading and a legend</h1></legend>    
    
    <label for="demo-input">Label</label>
    <input id="demo-input" type="text">
  </fieldset>
</form>
```

## What was the environment where this issue occurred?

* Device: PC
* Operating System: Windows 10
* Browser: Chrome
* Browser version: 94
* NHS.UK frontend package version: N/A - this is a basic html issue
* JAWS version - 2021.2107.12

**Comment by nickcolley** [comment]
It may vary between different assistive technologies it's been a while since I've tested this so can't remember I'll be honest. Worth trying more than JAWs for this one I think if you're able to.

**Comment by andymantell** [comment]
oh absolutely, I'm testing as much as I can get my hands on, and, though he doesn't know it yet, I'm hoping @mcheung-nhs might be able to have a look through as well for another pair of eyes

**Comment by andymantell** [comment]
Using the following markup pattern has tested well so far:

```
<legend class="nhsuk-fieldset__legend nhsuk-fieldset__legend--l" role="heading" aria-level="1">
  <span class="nhsuk-fieldset__heading">What is your address?</span>
</legend>
```

Tested in:

Windows 10 | Chrome 94 | JAWS 2021
Windows 10 | Chrome 94 | NVDA 2021.1
Windows 10 | Firefox | NVDA 2021.1 
Windows 10 | IE11 | NVDA 2021.1 
Windows 10 | Edge | NVDA 2021.1
Android 10 | Chrome 94 | TalkBack
iOS 14.7.1 | Safari | Voiceover

@mcheung-nhs Would you be able to take a look through the examples here https://nhs-duec.github.io/headings-inside-legends-test-cases/ at some point? Would be good to get another opinion. I'll DM you separately - it might be good to have a quick video call to explain.

**Comment by andymantell** [comment]
@adamliptrot-oc was able to test MacOS / Voiceover for me as I can't test that platform - but unfortunately it appears that the heading was not accessible when using the `role="heading"`. Which is probably enough to make this pattern unviable :-/

**Comment by andymantell** [comment]
Update from https://github.com/alphagov/govuk-frontend/issues/1605#issuecomment-971454440

> "We've had an update from Freedom Scientific which suggests that this bug will be looked at as part of their 2022 cycle (i.e: released as an update at some point to version 2022)."

**Comment by andymantell** [comment]
As per the update on this thread, this is now fixed in JAWS 2022:

https://github.com/alphagov/govuk-frontend/issues/1605#issuecomment-1077436947
